lib/instance_agent/plugins/codedeploy/command_executor.rb

```diff
@@ -473,6 +473,24 @@ def app_spec_path
           'appspec.yml'
         end
 
+        # Checks for existence the possible extensions of the app_spec_path (.yml and .yaml)
+        private
+        def app_spec_real_path(deployment_spec)
+          app_spec_yaml_location = File.join(archive_root_dir(deployment_spec), "appspec.yaml")
+          app_spec_yml_location = File.join(archive_root_dir(deployment_spec), "appspec.yml")
+          app_spec_default_location = File.join(archive_root_dir(deployment_spec), deployment_spec.app_spec_path)
+          if File.exist? app_spec_default_location
+            log(:debug, "BBBBB Using appspec file #{app_spec_default_location}")
+            app_spec_default_location
+          elsif File.exist? app_spec_yaml_location
+            log(:debug, "Using appspec file ${app_spec_yaml_location}")
+            app_spec_yaml_location
+          else
```

Comment:
would it be smart to also check whether yml exists? Then you could throw a descriptive error.

if yes, then:
(minor) About order - is the yaml or the yml location more likely?

Reply:
On the (minor) question you have:
- `yml` is the previously **hardcoded** location. 
- `yaml` is the last alternative

### Before 
Before this PR, we were just going for: `appspec.yml`, which is hardcoded.

### With this PR
we use, in the following order:
- `AppSpecPath` (a new variable you can specify via commandline). This will be the preferred location. If this param is not specified, it defaults to `appspec.yml` 
- if the file under `AppSpecPath` doesn't exist, the argument is not specified, we try with `appspec.yaml`
- if even the `appspec.yaml` doesn't exist, we try as last resort the `appspec.yml`

Reply:
I don't have a hard opinion, just pointing out that the behavior would now be changed for people who have both appspec.yml and appspec.yaml. That's why I though that the previous hardcoded value should come first (well, after the override, of course). Up to you.